adr/proposed/StatementSeperators.md

```diff
@@ -0,0 +1,111 @@
+# Overview
+
+Every SQL dialect has a way to separate:
+
+1. Statements
+2. Batches
+
+In most SQL dialects, the statement separator is a semi-colon (';'). In some SQL dialects, the statement separator can be dynamically configured.
+
+Further, some SQL dialects have the concept of _statement batches_. In these dialects, normally there are also "batch statement rules", such as:
+
+1. Some special statements must be the first statement in the batch, like `CREATE TABLE` statements.
+2. Some statements within the batch must be explicitly separated with a statement separator, like T-SQL Common Table Expressions. e.g., `;WITH CTE AS ( /* some sql */ ) SELECT * FROM CTE`
+
+# Database-Specific Statement Separators
+
+Below is a list of statement separators for every database FluentMigrator supports:
+
+<table>
+  <thead>
+    <tr>
+      <th> Database Type </th> <th> Default </th> <th> Override </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td> MySQL </td> <td> <code>;</code> </td>
+<td>
+
+```sql
+DELIMITER <char sequence>
+```
+
+</td>
+    </tr>
+    <tr>
+      <td> Postgres </td>
+<td> "dollar quoting" e.g.<br/><br/>
+
+```sql
+$function$
```

Comment:
doesn't Postgres have ";" as default separator ?

Reply:
You're right. I will update.